llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp

```diff
@@ -3741,9 +3741,11 @@ LegalizerHelper::lower(MachineInstr &MI, unsigned TypeIdx, LLT LowerHintTy) {
   }
   case TargetOpcode::G_ATOMIC_CMPXCHG_WITH_SUCCESS: {
     auto [OldValRes, SuccessRes, Addr, CmpVal, NewVal] = MI.getFirst5Regs();
-    MIRBuilder.buildAtomicCmpXchg(OldValRes, Addr, CmpVal, NewVal,
+    auto Tmp = MRI.createGenericVirtualRegister(MRI.getType(OldValRes));
```

Comment:
Should avoid creating temporary input regs like this

Reply:
Then it crashes, like I was trying to fix at the first place (https://github.com/llvm/llvm-project/pull/86357#discussion_r1536355410).